team_project_access.go

```diff
@@ -57,14 +58,105 @@ type TeamProjectAccessList struct {
 
 // TeamProjectAccess represents a project access for a team
 type TeamProjectAccess struct {
-	ID     string                `jsonapi:"primary,team-projects"`
-	Access TeamProjectAccessType `jsonapi:"attr,access"`
+	ID              string                `jsonapi:"primary,team-projects"`
+	Access          TeamProjectAccessType `jsonapi:"attr,access"`
+	ProjectAccess   *ProjectAccess2       `jsonapi:"attr,project-access"`
```

Comment:
ProjectAccess is already taken in type_helpers.go. Because it's not something that gets called directly in a client, and in order to not have to do breaking changes without the rest of the semver process, we opted to go with this.